### DISCUSSION TITLE
Container does not run with below error

### DISCUSSION THREAD

**Discussion by ahussaini01** [custompbx/custompbx#6] [General]
Hi 
I followed your instruction running docker container on Mac OS and I get below error:

24-02-07 14:11:23 Config file not found. Creating...
2024-02-07 14:11:23 2024/02/07 14:11:23 Config file at /app/config.json
2024-02-07 14:11:24 2024/02/07 14:11:24 CustomPBX development version: 0.0.105
2024-02-07 14:11:24 2024/02/07 14:11:24 Starting...
2024-02-07 14:11:24 2024/02/07 14:11:24 DB
2024-02-07 14:11:24 2024/02/07 14:11:24 Failed to connect to the database. Retrying in 5 seconds...
2024-02-07 14:11:29 2024/02/07 14:11:29 Failed to connect to the database. Retrying in 5 seconds...
2024-02-07 14:11:34 2024/02/07 14:11:34 Failed to connect to the database. Retrying in 5 second

I am unable to launch the customPBX GUI or anything.

Can you please helo?

**Comment by ahussaini01** [comment]
I also see this on Docker dashbaord: 2024-02-07 14:04:20.215399 0.00% [ERR] mod_xml_curl.c:319 Received HTTP error 0 trying to fetch https://127.0.0.1:8080/conf/config

**Comment by zusrut** [comment]
Hi. Obliviously cpbx cannot connect to the database. Make sure postgres container is up.
 The docker example made only for testing now. You do not need to change anything except `SIGNALWIRE_TOKEN` and `WS_BACKGROUND_OVERRIDE` to make it started.

**Comment by ahussaini01** [comment]
Thank for your reply
Postgres’s do work fine and my token is ok, I can login to Freeswitch with
no problem proving my token is fine too and Freeswitch is installed
successfully
My doubt is about WS_BACKGROUND_OVERRIDE
What should I put there ?

**Comment by zusrut** [comment]
`WS_BACKGROUND_OVERRIDE` is optional for change (not need to do it if you use test docker config) - it provides websocket url for frontend to connects to. In case frontend run separately for example.

**Comment by ahussaini01** [comment]
thanks and sorry for newbies questions.
once all 3 containers are running, presumable, I should be able to login to
GUI from my actual host machine running docker ? if yes what do I put in
brower?

I am puzzled how to get access to login page of GUI from host machine while
containers are running in Docker?

**Comment by zusrut** [comment]
If everything started right you just need to follow [https://127.0.0.1:8080/cweb]( https://127.0.0.1:8080/cweb) link and allow self signed certificates as it said in README file.

**Comment by ahussaini01** [comment]
thank you, I could successfully launch it by running the container in
different machine, not sure what was wrong on other machines
Thanks a Billion
[image: image.png]
